node/clients/masterdata.ts

```diff
@@ -0,0 +1,129 @@
+import { AuthenticationError, ForbiddenError, InstanceOptions, IOContext, JanusClient, RequestConfig, UserInputError } from '@vtex/api'
+import { AxiosError } from 'axios'
+import FormData from 'form-data'
+import { mergeAll, zipObj } from 'ramda'
+
+interface PaginationArgs {
+  page: number,
+  pageSize: number
+}
+
+interface Document {
+  Id: string
+  Href: string
+  DocumentId: string
+  [key: string]: any
+}
+
+interface KeyValue {
+  key: string
+  value: string
+}
+
+/*
+ * Convert a list of fields like [ {key: 'propertyName', value: 'String'}, ... ]
+ * to a JSON format.
+ */
+const parseFieldsToJson = (fields: KeyValue[]) => mergeAll(
+  fields.map((field: any) => zipObj([field.key], [field.value])),
+)
+
+const statusToError = (e: any) => {
+  if (!e.response) {
+    throw e
+  }
+  const { response } = e as AxiosError
+  const { status } = response!
+  if (status === 401) {
+    throw new AuthenticationError(e)
+  }
+  if (status === 403) {
+    throw new ForbiddenError(e)
+  }
+  if (status === 400) {
+    throw new UserInputError(e)
+  }
+  throw e
+}
+
+const paginationArgsToHeaders = ({page, pageSize}: PaginationArgs) => {
+  if (page < 1) {
+    throw new UserInputError('Smallest page value is 1')
+  }
+  const startIndex = (page - 1) * pageSize
+  const endIndex = startIndex + pageSize
+  return {
+    'REST-Range': `resources=${startIndex}-${endIndex}`
+  }
+}
+
+export class MasterData extends JanusClient {
+  constructor (ctx: IOContext, options?: InstanceOptions) {
+    super(ctx, {
+      ...options,
+      headers: {
+        ...options && options.headers,
+        ...{Accept: 'application/vnd.vtex.ds.v10+json'},
+        ...ctx.adminUserAuthToken ? {VtexIdclientAutCookie: ctx.adminUserAuthToken} : null,
+        ...ctx.storeUserAuthToken ? {[`VtexIdclientAutCookie_${ctx.account}`]: ctx.storeUserAuthToken} : null,
+      },
+    })
+  }
+
+  public getDocument = (acronym: string, id: string, fields: string[]) => this.get<Document>(
+    this.routes.document(acronym, id),
+    { metric: 'masterdata-getDocument', params: { _fields: fields } }
+  )
+
+  public createDocument = (acronym: string, fields: KeyValue[]) => this.post<Document>(
+    this.routes.documents(acronym),
+    { metric: 'masterdata-createDocument', params: parseFieldsToJson(fields) }
```

Comment:
this is wrong

Reply:
Why?

Reply:
it should use the data parameter on the post method to send the body like:
```
public createDocument = (acronym: string, fields: KeyValue[]) =>
    this.post<Document>(this.routes.documents(acronym),
    parseFieldsToJson(fields),
    { metric: 'masterdata-createDocument' }
)
```